### PR TITLE
Ingest script for UN SDG dataset

### PR DESCRIPTION
**Pull request by spoonerf**
This downloads the data for all 17 Sustainable Development Goals and for all geographic areas.

### CODE REVIEW

File: ingests/un_sdg.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import json
+import yaml
+import pandas as pd
+import tempfile
+import requests
+import datetime as dt
+import os
+
+from structlog import get_logger
```
Comment (Marigold) [inline ingests/un_sdg.py:9]: I've noticed `structlog` is not installed in walden, would you mind running `poetry add structlog` and committing changes here?

**Review by Marigold — APPROVED** [review]
Looks good! Couple of things could be DRYed with `worldbank_wdi.py`, but we don't have to do it now, it's just refactoring.

**Comment by Marigold** [comment]
Just a note that we already have two SDG datasets in ETL - one from [open numbers](https://github.com/owid/etl/blob/master/dag_files/dag_open_numbers.yml#L182) and second one [backported from our database](https://github.com/owid/etl/blob/master/dag.yml#L84) which reflects our data on https://sdg-tracker.org/. Still, I think it's better to have our version of SDG indicators end-to-end in ETL rather than having it from someone else or backported. Once we get SDG into ETL then maybe we could remove these two datasets that we already have.

File: ingests/un_sdg.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+
+def download_data() -> pd.DataFrame:
+    # retrieves all goal codes
+    print("Retrieving SDG goal codes...")
+    url = f"{base_url}/v1/sdg/Goal/List"
+    res = requests.get(url)
+    assert res.ok
+
+    goals = json.loads(res.content)
```
[the remark this answers is not in the document]
Comment (spoonerf) [reply ingests/un_sdg.py]: Ah thanks! Please don't hold back on the nitpicking, it's super helpful for me to learn!

**Comment by spoonerf** [comment]
> Just a note that we already have two SDG datasets in ETL - one from [open numbers](https://github.com/owid/etl/blob/master/dag_files/dag_open_numbers.yml#L182) and second one [backported from our database](https://github.com/owid/etl/blob/master/dag.yml#L84) which reflects our data on https://sdg-tracker.org/. Still, I think it's better to have our version of SDG indicators end-to-end in ETL rather than having it from someone else or backported. Once we get SDG into ETL then maybe we could remove these two datasets that we already have.

We definitely want to keep the backport of the SDG Tracker, I think this is a super valuable resource as it's a more curated version of the SDG indicators than the one I am importing here. Ideally we want to make the backport downloadable from sdg-tracker.org in the future. I agree about open numbers though!

**Pull request merged**